### PR TITLE
Extend CPU implementation of MSELoss to BF16

### PR DESCRIPTION
**Pull request by malfet**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #139960
* __->__ #139959
* #139961

It's strange that it has not been implemented for the type yet

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

### CODE REVIEW

File: aten/src/ATen/native/cpu/BinaryOpsKernel.cpp [hunk 1]
```diff
@@ -965,7 +965,7 @@ void mse_kernel(TensorIteratorBase& iter) {
         "This may be slower than using float or double-type tensors.");
   }
 
-  AT_DISPATCH_FLOATING_TYPES_AND_HALF(iter.dtype(), "mse_cpu", [&]() {
+  AT_DISPATCH_FLOATING_TYPES_AND2(kHalf, kBFloat16, iter.dtype(), "mse_cpu", [&]() {
```
Comment (jgong5) [inline aten/src/ATen/native/cpu/BinaryOpsKernel.cpp:968]: Do we have UT to cover the BF16 path?

**Review by jgong5 — APPROVED** [review]

File: aten/src/ATen/native/cpu/BinaryOpsKernel.cpp (hunk 1 above)
Comment (malfet) [reply aten/src/ATen/native/cpu/BinaryOpsKernel.cpp:968]: Good point, let's wait for CI signal, if it'll green(which it should not), I'll modify op_db to test on bf16

Comment (malfet) [reply aten/src/ATen/native/cpu/BinaryOpsKernel.cpp:968]: And there are, but for some reason they are not running on this PR, but on the followup, see https://github.com/pytorch/pytorch/actions/runs/11715233003/job/32631645140

File: torch/testing/_internal/common_methods_invocations.py
Comment (janeyx99) [inline torch/testing/_internal/common_methods_invocations.py:1]: looks right

**Review by janeyx99 — APPROVED** [review]

**Pull request closed without merging**